red4ext/src/downcast.rs

```diff
@@ -0,0 +1,172 @@
+//!
+//! It implies that you need definitions for:
+//! - [DelaySystem](https://github.com/WopsS/RED4ext.SDK/blob/master/include/RED4ext/Scripting/Natives/Generated/game/DelaySystem.hpp)
+//!   - let's use its method [DelayCallbackNextFrame](https://codeberg.org/adamsmasher/cyberpunk/src/branch/master/orphans.swift#L11540)
+//! - [DelayCallback](https://github.com/WopsS/RED4ext.SDK/blob/master/include/RED4ext/Scripting/Natives/Generated/game/DelaySystemScriptedDelayCallbackWrapper.hpp)
+//!
+//! First, define API in Redscript:
+//!
+//! ```swift,ignore
+//! module My
+//!
+//! class CustomCallback extends Callback {
+//!     private let message: String;
+//!     public func Call() -> Void {
+//!         LogChannel(n"DEBUG", s"received callback: \(this.message)");
+//!     }
+//! }
+//!
+//! public class System extends ScriptableSystem {
+//!     // dummy constructor for callback
+//!     func CreateCustomCallback(message: String) -> ref<CustomCallback> {
+//!         let callback: ref<CustomCallback> = new CustomCallback();
+//!         callback.message = message;
+//!         return callback;
+//!     }
+//! }
+//! ```
+//!
+//! Then define API wrappers with `red4ext-rs`:
+//!
+//! ```edition2021,ignore
+//! # use red4ext_rs::prelude::{Ref, RefRepr, RedString, IScriptable, Strong};
+//! # use red4ext_rs::downcast::{Downcast, TypedRef, FromTypedRef, IntoTypedRef};
+//! # use red4ext_rs::macros::redscript_import;
+//! // bind vanilla game classes
+//!
+//! #[derive(Default, Clone)]
+//! #[repr(transparent)]
+//! pub struct DelayCallback(Ref<IScriptable>);
+//!
+//! unsafe impl RefRepr for DelayCallback {
+//!     const CLASS_NAME: &'static str = "gameDelaySystemScriptedDelayCallbackWrapper";
+//!     type Type = Strong;
+//! }
+//!
+//! /// ✅ do not forget to implement `FromTypedRef<Self>`
+//! impl FromTypedRef<DelayCallback> for DelayCallback {
+//!     fn from_typed_ref(reference: TypedRef<Self>) -> Self {
+//!         Self(reference.into_inner())
+//!     }
+//! }
+//!
+//! #[derive(Default, Clone)]
+//! #[repr(transparent)]
+//! pub struct DelaySystem(Ref<IScriptable>);
+//!
+//! unsafe impl RefRepr for DelaySystem {
+//!     const CLASS_NAME: &'static str = "gameDelaySystem";
+//!     type Type = Strong;
+//! }
+//!
+//! #[redscript_import]
+//! impl DelaySystem {
+//!     /// ⬇️ please note how, here, method only accept parent class instance
+//!     #[redscript(native)]
+//!     pub fn delay_callback_next_frame(&self, callback: DelayCallback) -> ();
+//! }
+//!
+//! // bind custom classes
+//!
+//! #[derive(Default, Clone)]
+//! #[repr(transparent)]
+//! pub struct System(Ref<IScriptable>);
+//!
+//! unsafe impl RefRepr for System {
+//!     type Type = Strong;
+//!     const CLASS_NAME: &'static str = "My.System";
+//! }
+//!
+//! #[redscript_import]
+//! impl System {
+//!     /// import custom callback constructor
+//!     fn create_custom_callback(&self, message: RedString) -> CustomCallback;
```

Comment:
any reason why this doesn't just use `String`?
```suggestion
//!     fn create_custom_callback(&self, message: String) -> CustomCallback;
```